Synthetic float32 footers

corwinjoy opened the pull request:
I know that we primarily want real-world footers here but I believe there is also a place for some simple synthetic footers to zero-in on problem areas with performance.  As a start, I have contributed a simple set of float32 footers (since float32 tables are often used in numerical applications). 
This PR contains:
Synthetic footers using all float32 columns. Contents are: 1 rowgroup | 10,000 rows | float32 data type
with 10, 100, and 1000 columns.

corwinjoy committed:
Add a simple synthetic footer using all float32 columns. Contents are: 1 rowgroup | 10,000 rows | 1,000 columns | float32 data type

corwinjoy committed:
Add a simple synthetic float32 footers with 10 columns and 100 columns.

emkornfield commented:
Thanks for the contribution it might pay to add a information file to designate how these where generated.

alkis commented:
If we are going to take synthetic footers we should put them in `footer/synthetic` or some other directory. When one runs benchmarks against them it should be clear these are not realistic and shouldn't be "weighted" the same way as real ones.

Plus how are these generated? Do they Statistics, custom key-val data? From experience it is these auxiliary structures that mostly contribute to size and thus parsing speed.

rizaon commented:
Apache Impala has some unique synthetic parquet files that we use for unit tests.

Parquet files with deep nesting, used by [TestMaxNestingDepth](https://github.com/apache/impala/blob/da9400d63ce62e37ea531e7fe6564bf4ac2e0e45/tests/query_test/test_nested_types.py#L899)
https://github.com/apache/impala/blob/master/testdata/max_nesting_depth/README

[widetable_1000_cols](https://github.com/apache/impala/blob/da9400d63ce62e37ea531e7fe6564bf4ac2e0e45/testdata/datasets/functional/functional_schema_template.sql#L2451) table with 1000 columns parquet file.

[lineitem_sixblocks.parquet](https://github.com/apache/impala/blob/master/testdata/LineItemMultiBlock/lineitem_sixblocks.parquet) with 400 row groups.
https://github.com/apache/impala/blob/master/testdata/LineItemMultiBlock/README.dox

Are they interesting to take in?

alkis commented:
Apologies for the delay. I think they can be useful indeed. Let's add them with `synthetic-` prefix in the filename?

corwinjoy committed:
Add description of how and why data was generated.

corwinjoy commented:
@alkis Thanks for the feedback! Essentially, they are coming out of work on the parquet benchmarking project (https://github.com/Neelaksh-Singh/gresearch_parquet_benchmarking/tree/main) where we give more background and context. However, the main idea is to represent anonymized workloads that we see in building machine learning models at G-Research. Effectively, these are a large number of semi-random numeric columns where compression and dictionary coding don't have much effect.
I have updated the files to add a readme and some background code from the project. Also, I have renamed the files as suggested.
@Neelaksh-Singh

corwinjoy committed:
Remove old footer names.

alkis commented on footer/synthetic-test_data_generator.cc:
Let's remove this.

emkornfield replied:
I'd actually prefer to keep it, having source make things unambiguous.  I think putting these in there own folder makes more sense to me then the prefix.

alkis commented on footer/synthetic-readme.txt:
Can you instead add the three files above in footer/README.md along with the license you wish them to be shared with?

alkis commented on footer/synthetic-benchmark_float32_1000cols.footer:
Let's keep underscores for consistency and remove "benchmark" from the names? "benchmark" is already the name of the repo.

How about "synthetic_1000_float32_cols.footer" etc?
